in_toto/models/statement.py

```diff
@@ -0,0 +1,153 @@
+    the Predicate. Unset is treated the same as set-but-empty. May be 
+    omitted if predicate_type fully describes the predicate.
+  """
+  _type = attr.ib()
+  subject = attr.ib()
+  predicate_type = attr.ib()
+  predicate = attr.ib()
+  MODEL_NAME="Statement"
+
+
+  def __init__(self, **kwargs):
+    super(Statement, self).__init__()
+
+    self._type = STATEMENT_TYPE
+    self.subject = kwargs.get("subject")
+    self.predicate_type = kwargs.get("predicate_type")
+    self.predicate = kwargs.get("predicate", {})
+
+    self.validate()
+
+  @property
+  def type_(self):
+    """ The string to indentify the in-toto metadata type."""
+    # NOTE: We expose the type_ property in the API documentation instead of
+    # _type to protect it against modification.
+    # NOTE: Trailing underscore is used by convention (pep8) to avoid conflict
+    # with Python's type keyword.
+    return self._type
+
+  @staticmethod
+  def read(data):
+    """Creates a Statement object from its dictionary representation.
+
+    Arguments:
+      data: A dictionary with Statement metadata fields.
+
+    Raises:
+      securesystemslib.exceptions.FormatError: Passed data is invalid.
+
+    Returns:
+      The created Statement object.
+
+    """
+    return Statement(**data)
+
+
+  def _validate_type(self):
+    """Private method to check that `_type` is set correctly."""
+    if self._type != STATEMENT_TYPE:
+      raise securesystemslib.exceptions.FormatError(
+          "Invalid {}: field `_type` must be set to '{}', got: {}"
+          .format(self.MODEL_NAME, STATEMENT_TYPE, self._type))
+
+
+  def _validate_subject(self):
+    """Private method to check that `subject` is a `list` of `HASHDICTs`."""
+    if not isinstance(self.subject, list):
+      raise securesystemslib.exceptions.FormatError(
+          "Invalid {}: field `subject` must be of type dict, got: {}"
+          .format(self.MODEL_NAME, type(self.subject)))
+
+    for hash in self.subject:
+        for subject_key in hash:
+            subject_value = hash[subject_key]
+            if subject_key == "name":
+                if not isinstance(subject_value, str):
+                    raise securesystemslib.exceptions.FormatError(
+                    "Invalid {}: field `name` must be of type str, got: {}"
+                    .format(self.MODEL_NAME, type(subject_value)))
+
+                continue
+
+            securesystemslib.formats.HASHDICT_SCHEMA.check_match(subject_value)
+
+
+
+  def _validate_predicate_type(self):
+    """Private method to check that `predicate_type` is a `str`."""
+    if not isinstance(self.predicate_type, str):
+      raise securesystemslib.exceptions.FormatError(
+          "Invalid {}: field `predicate_type` must be of type str, got: {}"
+          .format(self.MODEL_NAME, type(self.predicate_type)))
+
+  def _validate_predicate(self):
+    """Private method to check that `predicate` is a `dict`."""
+    if not isinstance(self.predicate, dict):
```

Comment:
Perhaps we can skip this check for now. Once we define link predicate, slsa provenance etc, we'd want `self.predicate` to be an instance of one of those models rather than a dict.